File: biopython_rewrite/motif_search.py
#!/usr/bin/env python3
# Regex motif search in FASTA
import sys, re
from Bio import SeqIO
if len(sys.argv)<3:
    print('Usage: motif_search.py input.fasta motif_regex'); sys.exit(1)
infile = sys.argv[1]; motif = re.compile(sys.argv[2])
for rec in SeqIO.parse(infile,'fasta'):
    if motif.search(str(rec.seq)):
        print(rec.id)

File: legacy_perl/motif_search.pl
#!/usr/bin/env perl
# Search for regex motif in FASTA sequences
use strict; use warnings; use Bio::SeqIO;
my ($in,$motif) = @ARGV;
die "Usage: $0 input.fasta motif_regex\n" unless $in && $motif;
my $seqio = Bio::SeqIO->new(-file=>$in,-format=>'fasta');
while(my $r=$seqio->next_seq){
    if($r->seq =~ /$motif/){
        print $r->display_id, "\n";
    }
}
